apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -363,6 +393,34 @@ export class PnpmOptionsConfiguration extends PackageManagerOptionsConfiguration
    */
   public readonly useWorkspaces: boolean;
 
+  /**
+   * See https://pnpm.io/package_json#pnpmneverbuiltdependencies
+   *
+   * @remarks
+   * This field allows to ignore the builds of specific dependencies. The "preinstall", "install", and
+   * "postinstall" scripts of the listed packages will not be executed during installation.
+   */
+  public readonly neverBuiltDependencies: string[] | undefined;
+
+  /**
+   * See https://pnpm.io/package_json#pnpmpackageextensions
+   *
+   * @remarks
+   * The packageExtensions fields offer a way to extend the existing package definitions with additional
+   * information.
+   */
+  public readonly packageExtensions: IPnpmPackageExtensions | undefined;
```

Comment:
How does this interact with `.pnpmfile.cjs`?  Also why is it specified centrally for the monorepo?  It seems analogous to Yarn `resolutions` which is specified separately in the **package.json** for each project.

Reply:
If I'm understanding correctly, adding this is only relevant to when `useWorkspaces` is `false`, which is a behavior that we hope to retire in favor of delegating fully to the package manager. Otherwise this information should just be in package.json and get read by pnpm.

Reply:
Hi @octogonz ,

> How does this interact with .pnpmfile.cjs?

project manifest is handled in this order: overrides -> packageExtensions -> peerDependencyRules -> .pnpmfile.cjs readPackage

Detail: https://github.com/pnpm/pnpm/blob/a288467e6bff0e4a1f99ba78e2204bb1d48f031e/packages/core/src/install/index.ts#L498

> Also why is it specified centrally for the monorepo? 
> specified separately in the package.json for each project.

I believe it is intentional design to specify in a root project manifest(which is a top-level package.json in rush.json === `common/temp/package.json`)

As you can see here:

https://github.com/pnpm/pnpm/blob/a288467e6bff0e4a1f99ba78e2204bb1d48f031e/packages/plugin-commands-installation/src/recursive.ts#L126

pnpm internally call `recursive` for each project, and install in each project respects `pnpm.xxx` configurations from top-level package.json